src/main/java/com/google/step/youtube/YoutubeInfoScraper.java

```diff
@@ -27,52 +27,68 @@
       this.youTubeClient = youTubeClient;
   }
 
   public YoutubeInfoScraper() {
-    this(new YouTube.Builder(new NetHttpTransport(),
-        JacksonFactory.getDefaultInstance(),
-        /* httpRequestInitializer= */ null)
-        .setApplicationName(APPLICATION_NAME)
-        .setYouTubeRequestInitializer(new YouTubeRequestInitializer(API_KEY))
-        .build());
+      this(new YouTube.Builder(new NetHttpTransport(), JacksonFactory.getDefaultInstance(),
+              /* httpRequestInitializer= */ null).setApplicationName(APPLICATION_NAME)
+                      .setYouTubeRequestInitializer(new YouTubeRequestInitializer(API_KEY)).build());
   }
 
   /**
-  * @param channelId Id of a channel. Can be found in channel's urls.
-  * e.g. https://www.youtube.com/channel/<CHANNEL-ID>
-  * @return an optional string of the channel's upload playlist id. The optional will be empty 
-  * if id is invalid or no items were found.
-  */
-  public Optional<String> scrapeChannelUploadPlaylist(String channelId)
-    throws IOException {
-        ChannelListResponse response = youTubeClient.channels().list("contentDetails").setId(channelId).execute();
-        // getItems() return null when no items match the criteria (channelId). 
-        if (response.getItems()==null || response.getItems().isEmpty()) {
-            return Optional.empty();
-        }
-        List<Channel> channelsInfo = response.getItems();
-        // Since we are only requesting one channel-id, we only get one item back.  
-        return Optional.of(channelsInfo.get(0).getContentDetails().getRelatedPlaylists().getUploads());
-   }
+   * @param channelId Id of a channel. Can be found in channel's urls. e.g.
+   *                  https://www.youtube.com/channel/<CHANNEL-ID>
+   * @return an optional string of the channel's upload playlist id. The optional
+   *         will be empty if id is invalid or no items were found.
+   */
+  public Optional<String> scrapeChannelUploadPlaylist(final String channelId) throws IOException {
+      final ChannelListResponse response = youTubeClient.channels().list("contentDetails").setId(channelId).execute();
+      // getItems() return null when no items match the criteria (channelId).
+      if (response.getItems() == null || response.getItems().isEmpty()) {
+          return Optional.empty();
+      }
+      final List<Channel> channelsInfo = response.getItems();
+      // Since we are only requesting one channel-id, we only get one item back.
+      return Optional.of(channelsInfo.get(0).getContentDetails().getRelatedPlaylists().getUploads());
+  }
+
+  /**
+   * @param uploadId Id of a channel's upload playlist.
+   * @return an optional list of PlaylistItems. Each item contains a video-id,
+   *         description and a date. The optional will be empty if id is invalid
+   *         or no items were found.
+   */
+  public Optional<List<PlaylistItem>> scrapePlaylistItems(final String uploadId) throws IOException {
```

Comment:
optional: since this method is being called by `scrapePromoCodesFromPlaylist()` on line 79, it should be placed below `scrapePromoCodesFromPlaylist()`. Referencing Clean Code section on vertical formatting

Reply:
Fixed.